src/DataValues/DecimalValue.php

```diff
@@ -138,8 +122,8 @@ public function compare( self $that ) {
 			return 0;
 		}
 
-		$a = $this->getValue();
-		$b = $that->getValue();
+		$a = $this->value;
+		$b = $that->value;
```

Comment:
I worked extra hard to minimize the diff to an absolute minimum. I could have uploaded these 3 lines as a separate patch, and do this most of the time, but found its not worth it in this case.